module/app/ante/reject_extensions.go

```diff
@@ -66,20 +68,19 @@ func GetTxExtensionOptions(tx sdk.Tx) (hasExtensions bool, extensions []*sdkcode
 // Indicates if the given extension is a ExtensionOptionsWeb3Tx option, and returns the
 // unpacked option as well.
 // Otherwise returns (false, nil)
-func IsWeb3TxOption(cdc codec.Codec, extension *sdkcodec.Any) (eip712Signed bool, web3Extension *ethermint.ExtensionOptionsWeb3TxI) {
-	var web3Option ethermint.ExtensionOptionsWeb3TxI
+func IsWeb3TxOption(cdc codec.Codec, extension *sdkcodec.Any) (eip712Signed bool, web3Extension *ethermint.ExtensionOptionsWeb3Tx) {
 	if extension == nil { // invalid input
 		return false, nil
 	}
 	if extension.GetTypeUrl() != WEB3_EXTENSION_TYPE_URL {
 		return false, nil
 	}
-
-	if err := cdc.UnpackAny(extension, &web3Option); err != nil {
+	extOpt, ok := extension.GetCachedValue().(*ethermint.ExtensionOptionsWeb3Tx)
```

Comment:
are we sure the cached value is the same as unpacking?

Reply:
This is how it's done in the upstream ethermint repo, I'm merely applying the same approach.